pkg/grpc/server_test.go

```diff
@@ -62,6 +64,38 @@ func (a *APIServerSuite) TestEnvValues() {
 	}
 }
 
+func waitForPortToBeFree(t *testing.T, port uint64) {
+	ticker := time.NewTicker(10 * time.Millisecond)
+	defer ticker.Stop()
+
+	timer := time.NewTimer(10 * time.Second)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			listener, err := net.Listen("tcp4", fmt.Sprintf(":%d", port))
+			if err != nil {
+				t.Logf("Port %d still in use on tcp4", port)
+				continue
+			}
+			utils.IgnoreError(listener.Close)
+
+			listener, err = net.Listen("tcp6", fmt.Sprintf(":%d", port))
+			if err != nil {
+				t.Logf("Port %d still in use on tcp6", port)
+				continue
+			}
+			utils.IgnoreError(listener.Close)
```

Comment:
Do you think it would make sense to surface this error in case for whatever reason `Close` fails?